### ISSUE TITLE
The current code for ImageNet-1K only stores 1 replay buffer

### ISSUE THREAD

**Issue by ArmandXiao** [mchen725/DD_IGD#2]
Hi,

Thank you for the great work!

I am raising the issue to see if there is a misunderstanding about the method.
From the below code in `train_ckpts.py`, it does not store the **training trajectories**, but the **full 50 epochs** with different random network initializations.
```py
    for i in range(repeat):
        logger(f"Repeat: {i+1}/{repeat}")
        plotter = Plotter(args.save_dir, args.epochs, idx=i)
        model = define_model(args, nclass, logger)

        best_acc, acc, timestamps  = train(args, model, train_loader, val_loader, plotter, logger)
        trajectories.append(timestamps)
        best_acc_l.append(best_acc)
        acc_l.append(acc)

        # =================== Bugs? =========================
        if len(trajectories) == args.save_interval:
            n = int(args.start)
            while os.path.exists(os.path.join(args.ckpt_dir, "replay_buffer_{}.pt".format(n))):
                n += 1
            print("Saving {}".format(os.path.join(args.ckpt_dir, "replay_buffer_{}.pt".format(n))))
            torch.save(trajectories, os.path.join(args.ckpt_dir, "replay_buffer_{}.pt".format(n)))
            trajectories = []
         # =================================================

    # logger(f'\n(expert {repeat}) Best, last acc: {np.mean(best_acc_l):.1f} {np.std(best_acc_l):.1f}')
    print(f'\n(expert {repeat}) Best, last acc: {np.mean(best_acc_l):.1f} {np.std(best_acc_l):.1f}')
```

From both the descriptions of gradient-similarity-based checkpoint selection strategy, as well as the pre-computed checkpoints indices, it seems we should store the **training trajectories**.
```py
    elif  args.spec == '1k':
        if args.ckpt_path.startswith('ckpt'):
            if args.net_type == 'convnet6':
                idxs = [0,5,18,52]  # seems strange with only 50 training epochs
```

In addition, if we only train 50 epochs, does 52 here mean setting an interval smaller than one training epoch?

Can you help to recheck if the released checkpoint indices are correct (if so, what are the total steps?).

Many thanks in advance!
